pkg/sources/fake/fake.go

```diff
@@ -15,24 +15,19 @@ const (
 
 var startTime = time.Now()
 
-func getBattery(id uint16) msgs.TelemetryData {
+func getCAN(id uint16) msgs.CAN {
 	t := time.Since(startTime)
 	fid := float64(id)
-	voltage := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
-	temperature := float32(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
+	data := uint16(33000 + 3000*math.Sin((t.Seconds()/20)+fid))
 
-	return msgs.NewTelemetryData(&msgs.BatteryModule{
-		ID:          id,
-		Voltage:     voltage,
-		Temperature: temperature,
-	})
+	return msgs.NewCAN(id, data)
 }
 
 // GenFake generates fake data
 func GenFake(bus *pubsub.MessageBus) {
 	for {
-		for battery := uint16(1); battery <= 36; battery++ {
-			bus.Publish("battery", getBattery(battery))
+		for msg := uint16(1); msg <= 36; msg++ {
+			bus.Publish("CAN", getCAN(msg))
```

Comment:
Per-message/message is currently supported. Per-device would be possible but requires a complete map of the CAN ID and their corresponding originating devices. Because this is not currently done (or at least concrete enough) it is not currently in use. I would also like to implement a cross-platform original source for this information (ID, Message Meaning, Device) so that it can be cross compiled and used in Go, C (optionally don't know how critical it is there), Javascript and any other languages that may need it (Python for some scripts perhaps). I'm not sure the best platform for this ATM. If we don't need to support C parsing an ASCII protobuffer or JSON file with a map field is probably sufficient. But if we do want C support we may need a custom solution.

Subscribers or somewhere else downstream would implement the math to parse the messages. This is due to allow faster publishes and also because some subscribers (the database in particular) will store the data in raw form.